src/cloud-providers/ibmcloud-powervs/manager.go

```diff
@@ -31,6 +31,8 @@ func (_ *Manager) ParseCmd(flags *flag.FlagSet) {
 	flags.StringVar(&ibmcloudPowerVSConfig.ProcessorType, "proc-type", "shared", "Name of the processor type")
 	flags.StringVar(&ibmcloudPowerVSConfig.SystemType, "sys-type", "s922", "Name of the system type")
 	flags.BoolVar(&ibmcloudPowerVSConfig.UsePublicIP, "use-public-ip", false, "Use Public IP for connecting to the agent-protocol-forwarder inside the Pod VM")
+	flags.IntVar(&ibmcloudPowerVSConfig.PoolSize, "pool-size", 0, "Pool size of the pre-created pod VMs")
```

Comment:
The idea of a VM pool does not seem too IBM-cloud-specific to me. Can you put the VM pool changes in the common part of the code?

Reply:
Hi @c3d, yes I'm currently working on decoupling the pooling logic to make it provider agnostic.